core/dbt/contracts/graph/manifest.py

```diff
@@ -1031,10 +1051,11 @@ def add_node_nofile(self, node: ManifestNodes):
         _check_duplicates(node, self.nodes)
         self.nodes[node.unique_id] = node
 
-    def add_node(self, source_file: AnySourceFile, node: ManifestNodes):
+    def add_node(self, source_file: AnySourceFile, node: ManifestNodes, test_from=None):
         self.add_node_nofile(node)
         if isinstance(source_file, SchemaSourceFile):
-            source_file.tests.append(node.unique_id)
+            assert test_from
```

Comment:
Do we want to keep this assertion in? What is a failure mode that would trip this assertion?

Reply:
It's to keep mypy happy. Calling add_test without a test_from would fail. It shouldn't happen in practice, but mypy doesn't know that.